deploy-service/common/src/main/java/com/pinterest/deployservice/handler/PingHandler.java

```diff
@@ -822,4 +825,20 @@ public final static int getFinalMaxParallelCount(EnvironBean environBean, long t
         }
         return ret;
     }
+
+    public final static int getFinalMaxParallelCount(EnvironBean envBean, long totalNonFirstDeployAgents, int maxParallelThreshold) throws Exception{
+        return Math.min(maxParallelThreshold, getFinalMaxParallelCount(envBean, totalNonFirstDeployAgents));
+    }
+
+    public final static int getFinalMaxParallelCount(DeployConstraintBean deployConstraintBean, EnvironBean envBean, int totalHosts) throws Exception{
```

Comment:
where is this used?

Reply:
This is the extracted method name from the above question. The refactor was misplaced.

Reply:
plz ignore